app/importers/billing_contracts_import.rb

```diff
@@ -0,0 +1,53 @@
+class BillingContractsImport
+  include ActiveModel::Model
+  require 'roo'
+
+  attr_accessor :file
+
+  def initialize(attributes = {})
+    attributes.each { |name, value| send("#{name}=", value) }
+  end
+
+  def save
+    if imported_billing_contracts.map(&:valid?).all?
```

Comment:
why not just save if it is valid and if not then save errors somewhere you are reading this map of data twice

Reply:
totally agree already discussed on the last call that it should be rewritten
why didn't make any changes?